### PR TITLE
Update the autostart file when it's required

### LINKED ISSUE #788: Update the autostart file when required
In https://github.com/arduino/arduino-create-agent/pull/781 we moved the implementation of the auto-start feature of the agent directly in the agent itself. This was done not considering that there could be an old auto-start file already present on the user system. The auto-start file is not updated and still tries to run the old version of the Agent (or if the old version is manually removed, it runs nothing).
A solution would be to check if the auto-start file is up-to-date (instead of just checking if the file is present). If it is not updated, replace it with an updated version.

### PR DESCRIPTION
**Pull request by umbynos**
**Please check if the PR fulfills these requirements**

- [X] The PR has no duplicates (please search among the [Pull Requests](https://github.com/arduino/arduino-create-agent/pulls)
      before creating one)
- [ ] Tests for the changes have been added (for bug fixes / features)

* **What kind of change does this PR introduce?**

fix corner case introduced in #781 
- **What is the current behavior?**

Fix #788, the agent is now able to detect if the autostart file needs to be overwritten or not
* **What is the new behavior?**

The agent is now able to update the autostart file when it's required.
- **Does this PR introduce a breaking change?**

no
* **Other information**:

### CODE REVIEW

**Comment by umbynos** [comment]
Testable artifact here https://github.com/arduino/arduino-create-agent/releases/tag/1.2.8-rc12

**Review by cmaglie — APPROVED** [review]

File: config/autostart.go [hunk 1]
```diff
@@ -38,33 +39,56 @@ func getLaunchdAgentPath() *paths.Path {
...
+	} else {
+		// we already have an existing launchd plist file, so we check if it's updated
+		launchAgentContent, _ := launchdAgentPath.ReadFile()
+		launchAgentContentNew, _ := getLaunchdAgentDefinition()
+		if bytes.Equal(launchAgentContent, launchAgentContentNew) {
+			log.Infof("the autostart file %s already exists: nothing to do", launchdAgentPath)
+		} else {
+			log.Infof("the autostart file %s needs to be updated", launchdAgentPath)
+			removePlistFile()
+			writeLoadExit(launchdAgentPath)
```
Comment (cmaglie) [inline config/autostart.go:52]: In this case, `launchd` will not auto-load the app. I tried also to `launchctl unload` and `launchctl load` manually but no autostart of the app either. The effect is that you must run the app two times to get it start:
1. the first run will update the plsit file and exit
2. the second run will actually execute the app

Comment (umbynos) [reply config/autostart.go:52]: Ok, I will remove the `os.Exit(0)` in case the plist needs to be updated. This way should work

Comment (umbynos) [reply config/autostart.go]: should be solved by 0efac75

**Review by cmaglie — APPROVED** [review]
